src/BioFSharp/PWM.fs

```diff
@@ -0,0 +1,290 @@
+                    outerLoop tail (innerLoop acc [] head)
+            outerLoop il []
+
+        ///Checks whether all elements in the list have a wider distance than width or not.
+        let ceckForDistance (width:int) (items:int list) =
+            if items.Length <= 0 || items.Length = 1 then true
+            else
+                let rec loop n i =
+                    if n = items.Length-1 then true
+                    else
+                        if i >= items.Length then loop (n+1) (n+2)
+                        else
+                            if Operators.abs(items.[n]-items.[i]) > width then
+                                loop n (i+1)
+                            else false
+                loop 0 1
+
+        /// Get an integer which is between 0 and the length of the sequence - segmentLength
+        let getRandomNumberInSequence (segmentLength:int) (source:'a[]) =
+            let rnd = System.Random()
+            Array.init 1 (fun _ -> rnd.Next(0, source.Length-segmentLength+1))
+            |> Array.head
+
+        /// Create a specific sub sequence of the source sequence based on the given length and starting position. Do not forget, sequences start counting at 0!
+        let getDefinedSegment (subsequenceLength:int) (source:'a[]) (startPoint:int) =
+            source 
+            |> Array.skip startPoint 
+            |> Array.take subsequenceLength 
+            |> Array.ofSeq
+
+
+        ///Finds the best information content in an array of arrays of PWMSs and Positions.
+        let getBestInformationContent (item:((float*int)[])[]) =
+            let rec loop (n:int) (bestPWMS:(float*int)[]) =
+                if n = item.Length then bestPWMS
+                else
+                    let informationContentItem =
+                        item.[n]
+                        |> Array.fold (fun baseValue (pwms, _) -> pwms + baseValue) 0.
+                    let informationContentBestPWMS =
+                        bestPWMS
+                        |> Array.fold (fun baseValue (pwms, _) -> pwms + baseValue) 0.
+                    if informationContentItem > informationContentBestPWMS then
+                        loop (n + 1) item.[n]
+                    else
+                        loop (n + 1) bestPWMS
+            loop 0 [||]
+
+    open HelperFunctionsAndTypes
+
+    /// Contain tools to create and manipulate FrequencyVectors.
+    module FrequencyVector =
+
+        /// One dimensional array with fixed positions for each element.
+        /// Use to track frequency of elements independent of position in source.
+        type FrequencyVector internal (array:int []) =
+
+            inherit BaseArray<IBioItem, int>(array)
+
+            new() = 
+                let arr:'value [] = Array.zeroCreate 49
+                new FrequencyVector(arr)
+
+        /// Increase counter of position by 1.
+        let addInPlaceAtPos (bioItem:'a) (backGroundFrequencyVector:FrequencyVector) = 
```

Comment:
the naming is not quite on point here, better name it `increase` and the function below `increaseBy` instead of add